### PR TITLE
AmazonQ: Fix for incorrect code selection range values

### PR DESCRIPTION
**Pull request by ppiliaev**
## Problem

Intermittent 5xx errors are being returned from the backend due to invalid request payloads when customers have selected code on the first line of the file.

## Solution

The root cause was identified as incorrect payload formatting that occurred under specific conditions. The payload generation logic has been updated to properly handle these edge cases, and additional unit tests have been written to validate the payload structure in these scenarios

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dogusata — APPROVED** [review]

**Comment by hayemaxi** [comment]
Can you add to the description of this pr: the details of the edge cases and what the conditions are to reach these edge cases?

File: src/test/codewhispererChat/editor/context/focusArea/focusAreaExtractor.test.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+                line: 1,
+                character: 9,
+            },
+        }
+        const limit = 10
+
+        // Act
+        const focusAreaContextExtractor = new FocusAreaContextExtractor()
+        const method = Reflect.get(focusAreaContextExtractor, 'getExtendedCodeBlockRange')
+        const result = method.call(focusAreaContextExtractor, document, range, limit)
```
Comment (hayemaxi) [inline src/test/codewhispererChat/editor/context/focusArea/focusAreaExtractor.test.ts:115]: nit: is there a reason for this special syntax and not just
```
const focusAreaContextExtractor = new FocusAreaContextExtractor()
const result = focusAreaContextExtractor. getExtendedCodeBlockRange(document, range, limit)
```

**Review by hayemaxi — APPROVED** [review]

File: src/test/codewhispererChat/editor/context/focusArea/focusAreaExtractor.test.ts (hunk 1 above)
Comment (ppiliaev) [reply src/test/codewhispererChat/editor/context/focusArea/focusAreaExtractor.test.ts:115]: Yes, it's a private method, we cannot call it like this^
